### PR TITLE
#1380 Typescript conversion - View

### PR DESCRIPTION
**Pull request by emil-botha**
- Converted the View component to a functional component using TypeScript
- Implemented a unit test for the View component
- Remove selectable prop specified by Background and Badge
- Fix Avatar's AvatarContainer styled-component definition

### CODE REVIEW

File: packages/bappo-components/src/primitives/View/View.native.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    ref,
+  ) => {
+    const nativeViewRef = React.useRef<RN.View>(null);
+
+    React.useImperativeHandle(ref, () => ({
+      // To be able to use View inside TouchableHighlight/TouchableOpacity
+      setNativeProps: (...args: Array<any>) =>
+        nativeViewRef &&
+        nativeViewRef.current &&
+        nativeViewRef.current.setNativeProps({ ...args }),
```
Comment (williamfeng91) [inline packages/bappo-components/src/primitives/View/View.native.tsx]:
```suggestion
        nativeViewRef.current.setNativeProps(...args),
```

File: packages/bappo-components/src/primitives/View/View.native.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      testID,
+    }: ViewProps,
+    ref,
+  ) => {
+    const nativeViewRef = React.useRef<RN.View>(null);
+
+    React.useImperativeHandle(ref, () => ({
+      // To be able to use View inside TouchableHighlight/TouchableOpacity
+      setNativeProps: (...args: Array<any>) =>
+        nativeViewRef &&
```
Comment (williamfeng91) [inline packages/bappo-components/src/primitives/View/View.native.tsx]:
```suggestion
```
The ref object is never `null` or `undefined`.

**Review by williamfeng91 — CHANGES_REQUESTED** [review]

File: packages/bappo-components/src/primitives/View/View.native.tsx (hunk 1 above)
Comment (emil-botha) [reply packages/bappo-components/src/primitives/View/View.native.tsx]: Hi William, when I make the suggested change I get the following error:

image

Do I maybe have to change it to?:

image

Thanks!

Comment (williamfeng91) [reply packages/bappo-components/src/primitives/View/View.native.tsx]: Hi Emil. That looks good.

Comment (emil-botha) [reply packages/bappo-components/src/primitives/View/View.native.tsx]: Hi William, thanks, I've applied the change to `setNativeProps()`.

**Pull request merged**